### PR TITLE
LP-2938 - Integrate Delete Account, Reset Password & SSO Functionality on Account Page

### PR DESCRIPTION
**Pull request by aightmunam**
[LP-2938](https://philanthropyu.atlassian.net/browse/LP-2938)

[Theme PR](https://github.com/OmnipreneurshipAcademy/adg-edx-theme/pull/606)

***

#### Description:
- Add url for ADG account page i.e '/account/v2' 
- Add view for adg account page for

### CODE REVIEW

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/3260/workflows/workflows_id/jobs/3260/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/student/urls.py (100%)
openedx/adg/lms/student/views.py (100%)
-------------
Total:   6 lines
Missing: 0 lines
Coverage: 100%
-------------
```

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/3283/workflows/workflows_id/jobs/3283/artifacts)**. Coverage !!! :scream:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/student/urls.py (100%)
openedx/adg/lms/student/views.py (45.5%): Missing lines 24,26-28,37,42-44,46-48,61
-------------
Total:   25 lines
Missing: 12 lines
Coverage: 52%
-------------
```

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/3284/workflows/workflows_id/jobs/3284/artifacts)**. Coverage !!! :scream:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/student/urls.py (100%)
openedx/adg/lms/student/views.py (45.5%): Missing lines 24,26-28,37,42-44,46-48,61
-------------
Total:   25 lines
Missing: 12 lines
Coverage: 52%
-------------
```

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/3285/workflows/workflows_id/jobs/3285/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/student/tests/test_views.py (100%)
openedx/adg/lms/student/urls.py (100%)
openedx/adg/lms/student/views.py (90.9%): Missing lines 47-48
-------------
Total:   37 lines
Missing: 2 lines
Coverage: 94%
-------------
```

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/3286/workflows/workflows_id/jobs/3286/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/student/tests/test_views.py (100%)
openedx/adg/lms/student/urls.py (100%)
openedx/adg/lms/student/views.py (90.9%): Missing lines 47-48
-------------
Total:   37 lines
Missing: 2 lines
Coverage: 94%
-------------
```

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/3290/workflows/workflows_id/jobs/3290/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/student/tests/test_views.py (100%)
openedx/adg/lms/student/urls.py (100%)
openedx/adg/lms/student/views.py (81.8%): Missing lines 49-51,53
-------------
Total:   39 lines
Missing: 4 lines
Coverage: 89%
-------------
```

**Comment by aightmunam** [comment]
> **pylint:** ❌ **pycodestyle:** ✔️ **iSort:** ❌
> 
> **Please fix following pylint issues:**
> 
> ```
>  ************* Module edx-platform.openedx.core.djangoapps.user_api.accounts.views
> openedx/core/djangoapps/user_api/accounts/views.py:874:12: R1720: Unnecessary "elif" after "raise" (no-else-raise)
> openedx/core/djangoapps/user_api/accounts/views.py:1083:4: C0116: Missing function or method docstring (missing-function-docstring)
> openedx/core/djangoapps/user_api/accounts/views.py:1094:4: C0116: Missing function or method docstring (missing-function-docstring)
> openedx/core/djangoapps/user_api/accounts/views.py:56:0: W0611: Unused CourseEnrollment imported from common.djangoapps.student.models (unused-import)
> openedx/core/djangoapps/user_api/accounts/views.py:56:0: W0611: Unused is_username_retired imported from common.djangoapps.student.models (unused-import)
> openedx/core/djangoapps/user_api/accounts/views.py:114:0: I0021: Useless suppression of 'missing-module-docstring' (useless-suppression)
> openedx/core/djangoapps/user_api/accounts/views.py:114:0: I0021: Useless suppression of 'missing-class-docstring' (useless-suppression)
> openedx/core/djangoapps/user_api/accounts/views.py:114:0: I0021: Useless suppression of 'missing-function-docstring' (useless-suppression)
> openedx/core/djangoapps/user_api/accounts/views.py:577:0: I0021: Useless suppression of 'unused-argument' (useless-suppression)
> openedx/core/djangoapps/user_api/accounts/views.py:828:0: I0021: Useless suppression of 'unused-argument' (useless-suppression)
> 
> ------------------------------------------------------------------
> Your code has been rated at 9.91/10 (previous run: 9.91/10, +0.00) 
> ```
> 
> **Please fix following iSort issues:**
> 
> ```
>  ERROR: /openedx/core/djangoapps/user_api/accounts/views.py Imports are incorrectly sorted. 
> ```

The code quality workflow is failing due to the core change. Kindly ignore those.

**Review by annas-usman — APPROVED** [review]

File: openedx/core/djangoapps/user_api/accounts/views.py [hunk 1]
```diff
@@ -443,25 +444,27 @@ def post(self, request):
...
-                    notification = DeletionNotificationMessage().personalize(
-                        recipient=Recipient(username='', email_address=user_email),
-                        language=language_code,
-                        user_context=notification_context,
-                    )
-                    ace.send(notification)
-                except Exception as exc:
-                    log.exception('Error sending out deletion notification email')
-                    raise
+                if is_testing_environment():
```
Comment (asadali145) [inline openedx/core/djangoapps/user_api/accounts/views.py:447]: Is there any specific requirement to disable notifications? If not then we should not change the default flow.

**Review by asadali145 — APPROVED** [review]
Great work!

File: openedx/core/djangoapps/user_api/accounts/views.py (hunk 1 above)
Comment (aightmunam) [reply openedx/core/djangoapps/user_api/accounts/views.py:447]: I discussed this with Imran bhai and we decided we didn't need any for now.

**Pull request merged**